aiohttp/connector.py

```diff
@@ -1104,22 +1108,29 @@ async def _create_proxy_connection(
                             message=message,
                             headers=resp.headers,
                         )
-                    rawsock = transport.get_extra_info("socket", default=None)
-                    if rawsock is None:
-                        raise RuntimeError("Transport does not expose socket instance")
-                    # Duplicate the socket, so now we can close proxy transport
-                    rawsock = rawsock.dup()
-                finally:
+                except BaseException:
+                    # It shouldn't be closed in finally because it's fed to `loop.start_tls()`
+                    # and the docs say not to touch it after passing there.
                     transport.close()
-
-                transport, proto = await self._wrap_create_connection(
-                    self._factory,
-                    timeout=timeout,
-                    ssl=sslcontext,
-                    sock=rawsock,
+                    raise
+
+                setattr(
+                    transport, "_start_tls_compatible", True
```

Comment:
@asvetlov do you think it'd be okay to stick this conditionally somewhere in aiohttp or should we rather recommend the end-users to add this patch `setattr(asyncio.sslproto._SSLProtocolTransport, '_start_tls_compatible', True)` to their apps until the upstream CPython PR is accepted and shipped?

Reply:
I've decided to recommend keeping this monkey-patching in the user app.